readiness-tool/popup.js

```diff
@@ -0,0 +1,246 @@
+/** @const {string} */
+const blankMessage = '';
+/** @const {string} */
+const appsRegexDelimeter = '\\;';
+
+/**
+ * Callback function that sends a response upon receiving message
+ * @param {!Object} request - Message Object
+ * @param {!Object} sender - Message sender defined here
+ */
+chrome.runtime.onMessage.addListener(function (request, sender) {
+  if (request.action == 'getSource') {
+    // htmlOfTab contains the page source as a string
+    htmlOfTab = request.source;
+    findDetectedApps(htmlOfTab);
+  }
+});
+window.onload = function onWindowLoad() {
+  // When page is loaded, display 'Loading...' so the user expects content
+  supportedAds = document.getElementById('ads-supported');
+  supportedAnalytics = document.getElementById('analytics-supported');
+  notSupportedAds = document.getElementById('ads-notSupported');
+  notSupportedAnalytics = document.getElementById('analytics-notSupported');
+  supportedAds.textContent = supportedAnalytics.textContent =
+    notSupportedAds.textContent = notSupportedAnalytics.textContent =
+    loadingMessage;
+  // Gets the DOM of the current web page and converts it to a string
+  chrome.tabs.executeScript(null, {
+    file: 'getPagesSource.js',
+  }, function () {});
+};
+/**
+ * Returns all the 3rd party applications found on the website
+ * @param {string} html - String containing all HTML on the page
+ * @return {Object} 
+ */
+function findDetectedApps(html) {
+  fetch('apps.json').then(function (response) {
+      response.json().then(function (data) {
+        let listAllApps = data.apps;
+        detectedApps = filteredApps(html, listAllApps);
+        showSupportedAppsInView(detectedApps, listAllApps);
+        return detectedApps;
+      });
+    }).then(data => console.log('data is', data))
+    .catch(error => console.log('error is', error))
+}
+/**
+ * Add supported and unsupported applications to the view
+ * @param {Object} detectedApps - All 3rd Party Applications found on page
+ * @param {!Object} listAllApps - JSON of all 3p vendors
+ */
+function showSupportedAppsInView(detectedApps, listAllApps) {
+  supportedAds.textContent = supportedAnalytics.textContent =
+    notSupportedAds.textContent = notSupportedAnalytics.textContent =
+    blankMessage;
+  supportedAds.appendChild(makeList(detectedApps.supported.ads, false,
+    listAllApps));
+  supportedAnalytics.appendChild(makeList(detectedApps.supported.analytics,
+    false, listAllApps));
+  notSupportedAds.appendChild(makeList(detectedApps.notSupported.ads, true,
+    listAllApps));
+  notSupportedAnalytics.appendChild(makeList(
+    detectedApps.notSupported.analytics,
+    true, listAllApps));
+}
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {string} htmlString - String containing all HTML on the page
+ * @param {!Object} listAllApps - JSON of all the 3p vendors 
+ * @return {Object} 
+ */
+function filteredApps(htmlString, listAllApps) {
+  const foundThis = {
```

Comment:
`filteredApps`? It'd also be nice to have a detailed type declaration for this object or [`@typedef`](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#typedef-type) since it has a complex nested structure.